### ISSUE TITLE
`MATCH ... WHERE false` returns all nodes and has no effect

### ISSUE THREAD

**Issue by sapalli2989** [kuzudb/kuzu#2588]
```cypher
CREATE NODE TABLE T (id STRING, PRIMARY KEY(id));
CREATE (:T {id: "t1"}),(:T {id: "t2"});

kuzu> MATCH (t:T) WHERE false RETURN t;
---------------------------------
| t                             |
---------------------------------
| {_ID: 0:0, _LABEL: T, id: t1} |
---------------------------------
| {_ID: 0:1, _LABEL: T, id: t2} |
---------------------------------
(2 tuples)
```

Expected: `false` condition is never satisfied, hence `WHERE false` should lead to 0 results (compared with Neo4j).

Actual: All 2 results are returned.

Background: I was in search for a negated condition in combination with `CASE` given some specific user input value via prepared statement.

**Comment by andyfengHKU** [comment]
Hi @sapalli2989, the bug should now be fixed in PR #2590

**Comment by sapalli2989** [comment]
Much appreciated @andyfengHKU
